framework/include/utils/MooseMeshUtils.h

```diff
@@ -289,4 +290,182 @@ void makeOrderedNodeList(std::vector<std::pair<dof_id_type, dof_id_type>> & node
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void hexElemSplitter(ReplicatedMesh & mesh,
+                     const dof_id_type elem_id,
+                     std::vector<dof_id_type> & converted_elems_ids);
+
+/**
+ * Split a PYRAMID5 element into two TET4 element to prepare for the cutting operation.
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void pyramidElemSplitter(ReplicatedMesh & mesh,
+                         const dof_id_type elem_id,
+                         std::vector<dof_id_type> & converted_elems_ids);
+
+/**
+ * Split a PRISM6 element into three TET4 element to prepare for the cutting ope ration.
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void prismElemSplitter(ReplicatedMesh & mesh,
+                       const dof_id_type elem_id,
+                       std::vector<dof_id_type> & converted_elems_ids);
+
+/**
+ * Rotate a HEX8 element's nodes to ensure that the node with the minimum id is the first node;
+ * and the node among its three neighboring nodes with the minimum id is the second node.
+ * @param min_id_index The index of the node with the minimum id
+ * @param sec_min_pos The position of the node among its three neighboring nodes with the minimum
+ * id
+ * @param face_rotation A vector to record the rotation of the faces of the HEX8 element
+ */
+std::vector<unsigned int> nodeRotationHEX8(unsigned int min_id_index,
+                                           unsigned int sec_min_pos,
+                                           std::vector<unsigned int> & face_rotation);
+
+/**
+ * Calculate the three neighboring nodes of a node in a HEX8 element.
+ * @param min_id_index The index of the node with the minimum id
+ * @return a vector of the three neighboring nodes
+ */
+std::vector<unsigned int> neighborNodeIndicesHEX8(unsigned int min_id_index);
+
+/**
+ * For a vector of rotated nodes that can form a HEX8 element, create a series of four-node set
+ * that can form TET4 elements to replace the original HEX8 element. All the QUAD4 face of the
+ * HEX8 element will be split by the diagonal line that involves the node with the minimum id of
+ * that face.
+ * @param hex_nodes A vector of pointers to the nodes that can form a HEX8 element
+ * @param rotated_tet_face_indices A vector of vectors of the original face indices of the HEX8
+ * element corresponding to the faces of the newly created TET4 elements
+ * @return a vector of vectors of pointers to the nodes that can form TET4 elements
+ */
+std::vector<std::vector<Node *>>
+hexNodeOptimizer(std::vector<Node *> & hex_nodes,
```

Comment:
hexNodesToTetNodesSplit ?

optimizer sounds like a class

Reply:
May be determiner